### DISCUSSION TITLE
Base property of userConfig object does not prefix css imports

### DISCUSSION THREAD

**Discussion by rudyflores** [vuepress/core#751] [Q&A] [answered]
I am having this issue when deploying my project into GitHub pages, I am currently setting my object `base` value to the name of my repo for example `/my-project/` and have an import in the `head` property that links to a css stylesheet. but when running `vuepress build my-project` it seems to not prefix the css imports from `/assets/mystyles.css` to `/my-project/assets/mystyles.css`. Any idea if this may be a bug since I am currently using `vuepress 2.0.0-beta.36`?

**Comment by Mister-Hope** [comment, marked as answer]
**Minimal reproduction repo required** when you are taling about bugs, or at least please describe how you are importing that css. Do not start dicussion like this with little details.

And if you are importing that useing `head` option ,please note that we are **keeping them as is**. 

As an example, it's resonable for a person who want access to `/favicon.ico` when he is delopying some thing to a sub folder called `/docs/`

**Reply by rudyflores** [reply]
For my default config I am importing CSS inside the `head` property for example:
```typescript
base: `/github-user/my-repo/`

...

head: [
      [
                  "link",
                  {
                      rel: "stylesheet",
                      href: "/assets/css/custom.css"
                  }
      ]
]

```

but it seems to not append to the start of the href link the repo base name upon deployment

**Reply by rudyflores** [reply]
I also wanted to mention this is a typescript based config file and not a javascript based file

**Comment by Mister-Hope** [comment]
> And if you are importing that useing head option ,please note that we are keeping them as is.

> As an example, it's resonable for a person who want access to /favicon.ico when he is delopying some thing to a sub folder called /docs/

Google translate:

> 如果您使用 head 选项导入，请注意我们将它们保持原样。

> 例如，对于将某些内容部署到名为 /docs/ 的子文件夹人来说，想要访问 /favicon.ico 这是合理的。

Read it PLZ

**Reply by Mister-Hope** [reply]
Also , we never talked about we will do anything with absolute link (e.g.: pending base as prefix as you mentioned) for `head` options in docs. https://v2.vuepress.vuejs.org/reference/config.html#head

**Reply by rudyflores** [reply]
Sorry I think I understand what you mean, so your saying if the head property is used to attach my css files then they won't become appended with the repo link since it is intentional?

is there an alternative way to have these css files linked so they do become prefixed when deployed then?

**Reply by Mister-Hope** [reply]
We only mentioned here: https://v2.vuepress.vuejs.org/zh/guide/assets.html#base-helper and absolute link in markdown won't be pending with base either, bro.

**Reply by Mister-Hope** [reply]
> Sorry I think I understand what you mean, so your saying if the head property is used to attach my css files then they won't become appended with the repo link since it is intentional?
> 
> is there an alternative way to have these css files linked so they do become prefixed when deployed then?

You may need to append base yourself.

**Reply by Mister-Hope** [reply]
Or in another word, we only ensure that all the generate asset will have base prefix so that they can be obtained with correct url.

**Reply by rudyflores** [reply]
I see, is there a possibility you guys will ever add support for appending base to these kinds of css files in the future?

**Reply by Mister-Hope** [reply]
So I am very confused, which part of the docs is giving you the impressions that we will add base to absolute links **in project**

**Reply by Mister-Hope** [reply]
Since you are setting the base right besides head option in config, so I don' think there are actually any issue (or hard to append that for you)

It's not something we need to support. To many option with different behaviors will confuse users.

**Reply by rudyflores** [reply]
I see that makes sense, so would you recommend to either use a relative path instead for adding these css files instead since an absolute path seems to not be the right direction?

**Reply by Mister-Hope** [reply]
??? How can a relative path correct when you are in different pages, e.g: `/a.html` and `/a/b.html`?

**Reply by rudyflores** [reply]
I think im getting this mixed up from what you're saying, I understood relative paths will become prefixed since the documentation you lined seems to suggest using relative paths over absolute paths. In my case I need the absolute paths because I have a base property set to my gh-pages repo. if I wanted to use the `$withBase` variable with my project to prefix myself, how can I add my css links to my html file since the head is generated through the conifg?

**Reply by Mister-Hope** [reply]
Nope, the docs I provide is just telling you we are actually not pending prefix for absolute links in markdown files either.

Forget it if it's confusing you, I am just curious about the thought of pending base prefix.

We are keeping them as is, which means the url you fill in is the final url, so you should make them valid. So you should use `/base/link/to-your-css` if you want that url be correct. And as describe above, we will not adding any options to special deal with them (e.g.: adding base as prefix) , because  an option leading to total different behavior would confuse a lot of users.

**Reply by Mister-Hope** [reply]
Or in another word, since we donot mention in the docs that absolute link in proejct will be pended with base as prefix, you are supposed to not have confusion like this.
